keras/engine/training.py

```diff
@@ -1157,6 +1158,17 @@ def _fit_loop(self, f, ins, out_labels=None, batch_size=None,
         for cbk in callbacks:
             cbk.validation_data = val_ins
 
+        # To prevent a slowdown, we find beforehand the arrays that need conversion.
+        indices_for_conversion_to_dense = []
+        for i in range(len(self._feed_input_names)):
```

Comment:
There is a problem here. `_feed_input_names` is the list of model *inputs*, but `ins` refers to list of the Keras function input placeholders. Typically, ins = model_inputs + model_outputs + sample_weights. Your setup will still work, but the conversion doesn't get applied to targets.